avocado/plugins/multiplex.py

```diff
@@ -69,17 +69,16 @@ def configure(self, parser):
                                  help="Show the inherited values")
 
     def _activate(self, args):
-        # Extend default multiplex tree of --env values
+        # Extend default multiplex tree of --mux-inject values
         for value in getattr(args, "mux_inject", []):
-            value = value.split(':', 2)
+            value = value.split(':', 3)
             if len(value) < 2:
                 raise ValueError("key:value pairs required, found only %s"
                                  % (value))
-            elif len(value) == 2:
-                self._from_args_tree.value[value[0]] = value[1]
-            else:
-                node = self._from_args_tree.get_node(value[0], True)
-                node.value[value[1]] = value[2]
+            elif len(value) == 2:   # key, value
+                args.mux.data_inject(*value)
+            else:                   # path, key, value
+                args.mux.data_inject(value[1], value[2], value[0])
```

Comment:
With so many "value" mentions here, it becomes non-obvious to follow the parameters.  When using three parameters, it may be useful to name the parameters instead of using the comment.  IMHO, this:

```
args.mux.data_inject(key=value[1], value=value[2], path=value[0])
```

Is much more clear.  As mentioned earlier, changing the name of the `value` list (the result of the split) would help even further.